core/src/parser/utils.rs

```diff
@@ -539,17 +525,56 @@ where
 /// Merge two fields by performing the merge of both their value (dynamically, by introducing a
 /// merging operator) and their metadata (statically).
```

Comment:
Perhaps it's worth updating this comment to note that we don't always introduce the merge operator.